### DISCUSSION TITLE
Bug in configurationMap loading?

### DISCUSSION THREAD

**Discussion by jerchap** [nextgenhealthcare/connect#6055] [Q&A]
Ok, I think i found a bug.  I was recently having an issue where I would use configurationMap.get in a source transform, logged the value and also put it in the channel map, and observed that configurationMap.get was returning null.  I then decided to make a basic channel that lists all items in the configuration map and observed that sometimes it would only list a portion of the items, and sometimes it would list all the items.

Here is how I recreated the issue:
I created two channels (A and B)

Channel B is a channel reader, accepting JSON data type
Channel B has a single source transform with this code:

```
var i = 1;
for each(entry in configurationMap.entrySet().toArray()) {
    logger.info(i + ' Configuration ' + entry.getKey() + ' = ' + entry.getValue());
    i++;
}
```

Channel A is a file reader, type can be binary or text it doesn't matter.
Channel A has a single channel writer destination which sends a static json message to Channel B.  The template looks like this:

```
{
    "Test":"123"
}

```
​I have 30 items in my configuration map. 

If I drop a pdf (size doesn't seem to matter) for Channel A to pick up, it picks up, deletes the file and sends a static json message to Channel B, which then lists only 12 of the items in my configuration map.

If, through the Mirth dashboard I manually send a message to Channel A (instead of it picking up a file), it sends the static json message to Channel B, which then lists ALL the items in my configuration map.

I don't understand why the difference in how the message arrives in channel A would impact Channel B's ability to list configuration map items, especially because Channel A sends a static message to Channel B.

**Comment by pacmano1** [comment]
Can you try this instead and see if the value do all exist in the channelMap?

```js
var i = 1;
for each(entry in configurationMap.entrySet().toArray()) {
    $c(' Configuration_ ' + i, entry.getKey() + ' = ' + entry.getValue())
    i++;
}
```

**Reply by pacmano1** [reply]
(and if you post to the forums, cross reference the posts please).
